### ISSUE TITLE
How to deal with Dialects when Deparsing?

### ISSUE THREAD

**Issue by manticore-projects** [JSQLParser/JSqlParser#1157]
Provide Oracle specific dialect:
```sql

-- ORACLE ALTER TABLE ADD COLUMN
alter table risk.collateral
add  id_status              VARCHAR(1) NULL;

-- ORACLE ALTER TABLE MODIFY
alter table risk.collateral
MODIFY id_status              VARCHAR(1) NOT NULL;
```

Will deparse into Invalid Statements (under the scope of the Dialect):
```sql

-- ORACLE ALTER TABLE ADD COLUMN
ALTER TABLE risk.collateral ADD COLUMN id_status VARCHAR (1) NULL
;

-- ORACLE ALTER TABLE MODIFY
ALTER TABLE risk.collateral MODIFY COLUMN id_status VARCHAR (1) NOT NULL
;
```

So my questions are: Can I set/enforce a Dialect in JSQLParser? And how can I see/decide which Statements/Expressions are Dialect specific? (E. g. SELECT INTO comes into my mind.)

**Comment by manticore-projects** [comment]
As far as I can see you do expect the COLUMN keyword to appear:

```java 
    @Test
    public void testAlterTableAddColumn5() throws JSQLParserException {
        Statement stmt = CCJSqlParserUtil.parse("ALTER TABLE mytable ADD col1 timestamp (3)");

        // COLUMN keyword appears in deparsed statement
        assertStatementCanBeDeparsedAs(stmt, "ALTER TABLE mytable ADD COLUMN col1 timestamp (3)");

        Alter alter = (Alter) stmt;
        List<AlterExpression> alterExps = alter.getAlterExpressions();
        AlterExpression col1Exp = alterExps.get(0);
        List<ColumnDataType> col1DataTypes = col1Exp.getColDataTypeList();
        assertEquals("col1", col1DataTypes.get(0).getColumnName());
        assertEquals("timestamp (3)", col1DataTypes.get(0).getColDataType().toString());
    }
```

``` java
    @Test
    public void testAlterTableModifyColumn2() throws JSQLParserException {
        Statement stmt = CCJSqlParserUtil.parse("ALTER TABLE mytable modify col1 timestamp (6)");

        // COLUMN keyword appears in deparsed statement, modify becomes all caps
        assertStatementCanBeDeparsedAs(stmt, "ALTER TABLE mytable MODIFY COLUMN col1 timestamp (6)");

        assertEquals(AlterOperation.MODIFY, ((Alter) stmt).getAlterExpressions().get(0).
                getOperation());
    }
```

How is that supposed to work with Oracle?
If you agree, I would like to
* introduce a boolean property AlterExpression.oracleSpecific which is set to True only when ADD comes without COLUMN keyword or when the MODIFY keyword is set
* modify the JAVACC parser definition file to set this property during parsing
* amend the AlterExpression toString() method accordingly
* modify the Test Cases above.

What do you think please?

**Comment by manticore-projects** [comment]
Fixed by PR #1158, although I did not define any Dialects but only have checked of the COLUMN keyword has been used.
If not used we won't use it in the Deparser either.
I have corrected the test cases accordingly.

**Comment by wumpz** [comment]
JSqlParser per se has no option to enforce some kind of dialect. It tries to parse what it gets. You can afterwards build some kind of validation using the new validation framework of JSqlParser: https://github.com/JSQLParser/JSqlParser/wiki/Examples-of-SQL-Validation
